agent/hcp/client/telemetry_config.go

```diff
@@ -43,18 +44,14 @@ type MetricsConfig struct {
 	Labels   map[string]string
 	Filters  *regexp.Regexp
 	Endpoint *url.URL
+	Disabled bool
 }
 
 // RefreshConfig contains configuration for the periodic fetch of configuration from HCP.
 type RefreshConfig struct {
 	RefreshInterval time.Duration
 }
 
-// MetricsEnabled returns true if metrics export is enabled, i.e. a valid metrics endpoint exists.
```

Comment:
No need for this MetricsEnabeld func and the empty endpoint logic since we now have a `disabled` flag